### PR TITLE
HttpServer close handler not invoked if actual server is null

### LINKED ISSUE #2867: HttpServer close handler not invoked if actual server is null
If an exception occurs after `listening` is switched on but before `actualServer` is set, then the close method handler will not be invoked.

This prevents the `deploy` method to receive the failure (the verticle is undeployed correctly but the HTTP server hook never completes). Here's an example:

```java
public class HTTPSVerticle extends AbstractVerticle {

  @Override
  public void start(Future<Void> startFuture) {
    PfxOptions pfxOptions = new PfxOptions().setValue(Buffer.buffer("path")).setPassword("pwd");

    HttpServerOptions httpServerOptions = new HttpServerOptions();
    httpServerOptions
      .setSsl(true)
      .setPfxTrustOptions(pfxOptions);

    vertx.createHttpServer(httpServerOptions).requestHandler(req -> {
      req.response().end("Hello World!");
    }).listen(8443, ar -> {
      if (ar.succeeded()) {
        startFuture.complete();
      } else {
        startFuture.fail(ar.cause());
      }
    });
  }
}
```

If the PFX path is wrong, `listen` throws an exception.

```java
vertx.deploy(new HTTPSVerticle(), handler);
```

But the handler will not be invoked

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #2867

### CODE REVIEW

**Comment by tsegismont** [comment]
@vietj can you please review? I set the issue milestone to 3.7, if you're fine with backporting.

File: src/test/java/io/vertx/core/http/HttpServerCloseHookTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      complete();
+    }));
+    await();
+  }
+
+  private static class TestVerticle extends AbstractVerticle {
+
+    @Override
+    public void start(Future<Void> startFuture) {
+      PfxOptions pfxOptions = new PfxOptions().setValue(Buffer.buffer("boom"));
```
Comment (vietj) [inline src/test/java/io/vertx/core/http/HttpServerCloseHookTest.java]: rename to _invalidOptions_ so it's clearer it's not correct (took some time to figure this it)

Comment (tsegismont) [reply src/test/java/io/vertx/core/http/HttpServerCloseHookTest.java]: I thought `boom` was clear enough :stuck_out_tongue_winking_eye: 

But right, I'll update and merge. Thanks for reviewing.

**Comment by vietj** [comment]
I missed it because I overlooked the content value

**Pull request merged**